src/pepr/operator/controllers/network/generators/kubeNodes.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright 2024 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { KubernetesListObject } from "@kubernetes/client-node";
+import { V1NetworkPolicyPeer, V1NodeAddress } from "@kubernetes/client-node";
+import { K8s, kind, R } from "pepr";
+
+import { Component, setupLogger } from "../../../../logger";
+import { RemoteGenerated } from "../../../crd";
+import { anywhere } from "./anywhere";
+import { UDSConfig } from "../../../../config";
+import { retryWithDelay } from "../../utils";
+
+const log = setupLogger(Component.OPERATOR_GENERATORS);
+
+// Maintain a set of all node internal IPs
+const nodeSet = new Set<string>();
+
+/**
+ * Initialize the node targets by fetching the current nodes in the cluster
+ * and populating the nodeSet with their Internal IPs.
+ */
+export async function initAllNodesTarget() {
+  // if a list of CIDRs is defined, use those
+  if (UDSConfig.kubeNodeCidrs) {
+    const nodeCidrs = UDSConfig.kubeNodeCidrs.split(",");
+    for (const nodeCidr of nodeCidrs) {
+      nodeSet.add(nodeCidr);
+    }
+    return;
+  }
+
+  try {
+    const nodes = await retryWithDelay(fetchKubernetesNodes, log);
+    nodeSet.clear();
+
+    for (const node of nodes.items) {
+      const ip = getNodeInternalIP(node);
+      if (ip) nodeSet.add(ip);
+    }
```

Comment:
Similar suggestion here - assuming the fetch doesn't error we should update netpols as well.
```suggestion
    }
    await updateKubeNodesNetworkPolicies();
```